napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1620 @@
+
+            if type(edge_color) is str:
+                self._edge_color = edge_color
+            else:
+                self._edge_color = 'black'
+
+            if type(face_color) is str:
+                self._face_color = face_color
+            else:
+                self._face_color = 'white'
+            self._opacity = opacity
+
+            # update flags
+            self._need_display_update = False
+            self._need_visual_update = False
+
+            self._selected_shapes = []
+            self._selected_shapes_stored = []
+            self._selected_shapes_history = []
+            self._selected_box = None
+
+            self._hover_shape = None
+            self._hover_shape_stored = None
+            self._hover_vertex = None
+            self._hover_vertex_stored = None
+            self._moving_shape = None
+            self._moving_vertex = None
+
+            self._drag_start = None
+            self._fixed_vertex = None
+            self._fixed_aspect = False
+            self._aspect_ratio = 1
+            self._is_moving = False
+            self._fixed_index = 0
+            self._is_selecting = False
+            self._drag_box = None
+            self._drag_box_stored = None
+            self._cursor_coord = np.array([0, 0])
+            self._is_creating = False
+            self._update_properties = True
+            self._clipboard = []
+
+            self._mode = Mode.PAN_ZOOM
+            self._mode_history = self._mode
+            self._status = str(self._mode)
+            self._help = 'enter a selection mode to edit shape properties'
+
+            self.events.add(mode=Event,
+                            edge_width=Event,
+                            edge_color=Event,
+                            face_color=Event)
+
+            self._qt_properties = QtShapesLayer(self)
+            self._qt_controls = QtShapesControls(self)
+
+            self.events.deselect.connect(lambda x: self._finish_drawing())
+
+    @property
+    def data(self):
+        """ShapeList: object containing all the shape data
+        """
+        return self._data
+
+    @data.setter
+    def data(self, data):
+        self._data = data
+        self.refresh()
+
+    @property
+    def edge_width(self):
+        """float: width of edges in px
+        """
+        return self._edge_width
+
+    @edge_width.setter
+    def edge_width(self, edge_width):
+        self._edge_width = edge_width
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_edge_width(i, edge_width)
+            self.refresh()
+        self.events.edge_width()
+
+    @property
+    def edge_color(self):
+        """str: color of edges and lines
+        """
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        self._edge_color = edge_color
+        if self._update_properties is True:
```

Comment:
Just `self._update_properties` I think?

Reply:
(Same in other places with this pattern)

Reply:
fixed here and elsewhere